[WIP] Master shade

Akm0d opened the pull request:
### What does this PR do?
Add shade library so that openstack tests run properly on jenkins
Fixes all the problems with the openstack driver now that we can see the test failures
Fix all cloud test failures

### What issues does this PR fix or reference?

### Tests written?

Yes

### Commits signed with GPG?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

s0undt3ch commented on requirements/static/py2.7/osx.txt:
```diff
@@ -138,3 +138,7 @@ yamlordereddictloader==0.4.0
 zc.lockfile==1.4          # via cherrypy
 # Passthrough dependencies from pkg/osx/req.txt
 pyobjc==5.1.2
+# Passthrough dependencies from pkg/osx/req.txt
```
We apparently have a bug in our compile requirements code.
Once is enough :)

Akm0d committed:
Added openstack libraries to requirements

Akm0d committed:
Refactored cloud tests to use `assertCreateInstance`
  - Refactored virtualbox tests to new standard
  - Use the given timeouts for each test
  - Consolidated `test_cloud` into digital ocean test
  - Various stability improvements to the cloud tests

Akm0d committed:
Fixed linter errors and removed redundnat files

Akm0d committed:
Fixed openstack profile naming
